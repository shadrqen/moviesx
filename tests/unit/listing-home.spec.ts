import { shallowMount } from '@vue/test-utils'
import ListingHomePage from '@/views/ListingHome.vue'

describe('ListingHome', () => {
  it('test', () => {
    expect(1).toBe(1)
  })
})

/*
import { createLocalVue, shallowMount } from '@vue/test-utils'
import ListingHomePage from '@/views/ListingHome.vue'
import Vuetify from 'vuetify'
import flushPromises from 'flush-promises'
import axios from 'axios'

const localVue = createLocalVue()

localVue.use(Vuetify)

// Notice how `create` was not being mocked here...
jest.mock('axios')

describe('ListingHome', () => {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  let wrapper: any
  beforeEach(() => {
    wrapper = shallowMount(ListingHomePage, {
      localVue
    })
  })
  it('Fetches movies from the mock api', async () => {
    await flushPromises()
    expect(axios.create).toHaveBeenCalled()

    const table = wrapper.find('#main-table')

    expect(table.exists()).toBe(true)
  })

  it('should match snapshot', () => {
    expect(wrapper.element).toBeDefined()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('Should search for a movie', async () => {
    const searchTextField = wrapper.find('#search-text-field')

    searchTextField.element.value = 'Aquaman'

    expect(searchTextField.exists()).toBe(true)

    expect(wrapper.find('#search-text-field').element.value).toBe('Aquaman')
  })
})
*/
