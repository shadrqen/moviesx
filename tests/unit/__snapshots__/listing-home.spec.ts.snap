// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingHome should match snapshot 1`] = `
<v-container-stub
  class="listing-home-container"
  tag="div"
>
  <v-card-stub
    elevation="1"
    loaderheight="4"
    tag="div"
  >
    <v-card-title-stub
      tag="div"
    >
      <div>
        
        My Favourites
      
      </div>
       
      <v-spacer-stub
        tag="div"
      />
       
      <v-text-field-stub
        appendicon="mdi-magnify"
        backgroundcolor=""
        clearicon="$clear"
        errorcount="1"
        errormessages=""
        hidedetails="true"
        id="search-text-field"
        label="Search"
        loaderheight="2"
        messages=""
        rules=""
        singleline="true"
        successmessages=""
        type="text"
      />
       
      <v-spacer-stub
        tag="div"
      />
       
      <v-btn-stub
        activeclass=""
        class="mt-4"
        id="favourites-btn"
        outlined="true"
        tag="button"
        type="button"
      >
        <div>
          
          Show All
        
        </div>
      </v-btn-stub>
    </v-card-title-stub>
     
    <v-data-table-stub
      customfilter="[Function]"
      customgroup="[Function]"
      customsort="[Function]"
      expanded=""
      expandicon="$expand"
      groupby=""
      groupdesc=""
      headers=""
      id="main-table"
      itemclass=""
      itemkey="id"
      items=""
      itemsperpage="10"
      loaderheight="4"
      loadingtext="Loading... Please wait"
      locale="en-US"
      mobilebreakpoint="600"
      nodatatext="$vuetify.noDataText"
      noresultstext="$vuetify.dataIterator.noResultsText"
      options="[object Object]"
      page="1"
      selectablekey="isSelectable"
      serveritemslength="-1"
      sortby=""
      sortdesc=""
      value=""
    />
  </v-card-stub>
   
  <div
    class="text-center"
  >
    <v-snackbar-stub
      contentclass=""
      multiline="true"
      tag="div"
      timeout="5000"
      transition="v-snack-transition"
    >
      
      

      
    </v-snackbar-stub>
  </div>
   
  <v-overlay-stub
    color="#212121"
    dark="true"
    opacity="0.9"
    value="true"
    zindex="5"
  >
    <div
      class="lds-ellipsis"
    >
      <div />
       
      <div />
       
      <div />
       
      <div />
    </div>
  </v-overlay-stub>
</v-container-stub>
`;
